Objects/typeobject.c

```diff
@@ -4840,9 +4853,10 @@ PyType_GetModuleByDef(PyTypeObject *type, PyModuleDef *def)
     // mro_invoke() ensures that the type MRO cannot be empty, so we don't have
     // to check i < PyTuple_GET_SIZE(mro) at the first loop iteration.
     assert(PyTuple_GET_SIZE(mro) >= 1);
+    assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)type);
```

Comment:
Let's add a comment about why the loop starts at one.

The comment about "first loop iteration" should have been removed when Victor accidentally changed the MRO walk from a do-while loop to a for; let's remove it.
```suggestion
    // mro_invoke() ensures that the type MRO cannot be empty.
    assert(PyTuple_GET_SIZE(mro) >= 1);
    // Also, the first item in the MRO is the type itself, which
    // we already checked above. We skip it in the loop.
    assert(PyTuple_GET_ITEM(mro, 0) == (PyObject *)type);
```